### PR TITLE
Js fe values without dim

### PR DESCRIPTION
**Pull request by jstebel**
Finished simplification of FEValues (#1136). Main change is the removal of template parameter "dim" and unification of FEValues, FESideValues, FEValuesSpaceBase and FEValuesBase. Further remarks:
- Methods reinit() in ElementValues and FEValues take ElementAccessor/Side as argument instead of DHCellAccessor/DHCellSide. This is because we do not need to work with DOFHandler and there is problem on boundary elements, where DOFHandler cannot be used so far.
- FEValues does not store pointer to FiniteElement<dim> (because the template parameter dim is not available). Instead it saves necessary data such as number of components, structure of dofs etc.

### CODE REVIEW

File: src/fem/fe_values.hh [hunk 1]
```diff
@@ -245,7 +219,7 @@ public:
      *
      * @param point_no Number of the quadrature point.
      */
-	inline arma::vec::fixed<spacedim> normal_vector(unsigned int point_no) override
+	inline arma::vec::fixed<spacedim> normal_vector(unsigned int point_no)
```
Comment (jbrezmorf) [inline src/fem/fe_values.hh]: To discuss: Normal has no sense for the bulk quadrature points.

Comment (jstebel) [reply src/fem/fe_values.hh]: Should be resolved together with new mechanism of cached values and point sets.

File: src/fem/fe_values.hh [hunk 2]
```diff
@@ -284,25 +258,20 @@ public:
...
-
-    /**
-     * @brief Returns the finite element in use.
-     */
-    inline FiniteElement<dim> * get_fe() const
-    {
-        return fe;
-    }
+    /// Return dimension of reference space.
+    inline unsigned int dm() const
```
[the remark this answers is not in the document]
Comment (jstebel) [reply src/fem/fe_values.hh]: Done.

File: src/fem/fe_values.hh [hunk 3]
```diff
@@ -45,115 +45,89 @@ template<unsigned int dim> class FiniteElement;
...
+    };
   
 public:
 
-    /**
-     * @brief Default constructor
-     */
-    FEValuesBase();
+    /// Default constructor.
+    FEValues();
```
[the remark this answers is not in the document]
Comment (jstebel) [reply src/fem/fe_values.hh]: Done.

**Pull request merged**